sci-electronics/iverilog: add iverilog-9999.ebuild

vowstar opened the pull request:
* Found a better way to fix the build race condition, fixed in this ebuild
* Removed ``` .NOTPARALLEL: install``` to speed up build, and workaround parallel build bug
* Applied these fix in iverilog-10.3.ebuild

Upstream fix by @minux: https://github.com/steveicarus/iverilog/pull/294
Issue related: https://github.com/gentoo/gentoo/pull/14096

Package-Manager: Portage-2.3.84, Repoman-2.3.20
Signed-off-by: Huang Rui <vowstar@gmail.com>

juippis commented on sci-electronics/iverilog/iverilog-10.3.ebuild:
```diff
@@ -57,8 +54,20 @@ src_prepare() {
+	# NOTE: this is only a workaround for upstream parallel build bug
+	# this fix made by minux https://github.com/minux
+	# https://github.com/steveicarus/iverilog/pull/294
+	# Due to this issue, we must run ``emake installdirs`` before ``default``
```
Too much text here, the first line and upstream issue link is enough. Ie, 
```
# Default build fails with parallel jobs, 
# https://github.com/steveicarus/iverilog/pull/294
```
That's enough information.

vowstar replied:
Thank you! I've applied it to both 10.3 and 9999 ebuild

juippis commented on sci-electronics/iverilog/iverilog-10.3.ebuild:
```diff
@@ -57,8 +54,20 @@ src_prepare() {
+	# this fix made by minux https://github.com/minux
+	# https://github.com/steveicarus/iverilog/pull/294
+	# Due to this issue, we must run ``emake installdirs`` before ``default``
+	emake installdirs DESTDIR="${D}" || die 'emake failed'
```
`emake` has a built-in `die` function, so you're duplicating code here. These comments apply to -9999 ebuild as well.

Rest looks good and glad you found a solution to parallel install so fast!

vowstar replied:
> `emake` has a built-in `die` function, so you're duplicating code here. These comments apply to -9999 ebuild as well.
> 
> Rest looks good and glad you found a solution to parallel install so fast!

Thank you! I've remove `` || die 'emake failed'`` in both 10.3 and 9999 ebuild

juippis commented:
I split your commit into two, please test that 10.3 still works. I removed `src_compile() { default }` because it should be the default behaviour without defining it.

vowstar commented:
> I split your commit into two, please test that 10.3 still works. I removed `src_compile() { default }` because it should be the default behaviour without defining it.

Thank you very much!

I tested, after removed ``src_compile() { default }``, both 10.3 and 9999 are works. 
The ``install`` target in Makefile run the ``all`` target correctly.
